server/src/main/java/com/linecorp/centraldogma/server/plugin/PluginInitContext.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.plugin;
+
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.centraldogma.server.CentralDogmaConfig;
+
+/**
+ * A context that is used to pass when calling {@link AllReplicasPlugin#init(PluginInitContext)}.
+ */
+public final class PluginInitContext {
```

Comment:
Question) I was thinking we would need at least `ProjectManager`, `CommandExecutor` will be needed to build services which can write to `CentralDogma` storage. Is there no need for these values? (it looks like they are available at init time).

I actually expect we may also want access to `MeterRegistry`, and actually all values in `PluginContext` eventually.

Reply:
Oops, I thought somehow they are not created when the init method is called. 😅 
Let me fix that. 😉

Reply:
Fixed. PTAL. 😉